vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -1277,21 +1351,114 @@ def fused_experts(hidden_states: torch.Tensor,
                   w2_zp: Optional[torch.Tensor] = None,
                   a1_scale: Optional[torch.Tensor] = None,
                   a2_scale: Optional[torch.Tensor] = None,
-                  block_shape: Optional[List[int]] = None) -> torch.Tensor:
+                  block_shape: Optional[List[int]] = None,
+                  allow_deep_gemm: bool = True) -> torch.Tensor:
 
     if inplace:
         torch.ops.vllm.inplace_fused_experts(
             hidden_states, w1, w2, topk_weights, topk_ids, activation,
             use_fp8_w8a8, use_int8_w8a16, use_int4_w4a16, global_num_experts,
             expert_map, w1_scale, w2_scale, w1_zp, w2_zp, a1_scale, a2_scale,
-            block_shape)
+            block_shape, allow_deep_gemm)
         return hidden_states
     else:
         return torch.ops.vllm.outplace_fused_experts(
             hidden_states, w1, w2, topk_weights, topk_ids, activation,
             use_fp8_w8a8, use_int8_w8a16, use_int4_w4a16, global_num_experts,
             expert_map, w1_scale, w2_scale, w1_zp, w2_zp, a1_scale, a2_scale,
-            block_shape)
+            block_shape, allow_deep_gemm)
+
+
+def _fp8_perm(m: torch.Tensor, idx: torch.Tensor) -> torch.Tensor:
+    """
+    A permutation routine that works on fp8 types.
+    """
+    if torch.is_floating_point(m) and torch.finfo(m.dtype).bits == 8:
+        return m.view(dtype=torch.uint8)[idx,
+                                         ...].view(dtype=torch.float8_e4m3fn)
```

Comment:
for the second view, do we need to use `m.dtype` instead of `torch.float8_e4m3fn`?

Reply:
Good catch. Thanks!